src/main/java/uk/gov/hmcts/reform/pip/data/management/models/location/LocationMetadata.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.hmcts.reform.pip.data.management.models.location;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.util.UUID;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+@Entity
+@Table
+public class LocationMetadata {
```

Comment:
I think we should add some more validation on here, to fail if only a locationId is provided, and none of the other details.

I think in practice the three scenario's should be:

1) Caution Messages Only
2) NoList Messages Only
3) Caution and NoList Messages

Reply:
as discussed, add custom controller.